### ISSUE TITLE
Like writing Specs? Yes !!! How can I do it?

### ISSUE THREAD

**Issue by EugeneGirsh** [angular-ui/angular-google-maps#1282]
I didn't found any info how to begin. Pls, write to me

**Comment by nmccready** [comment]
I'll get back to you. But first read up on angular testing and jasmine. Otherwise look at existing specs.

**Comment by EugeneGirsh** [comment]
Thank you. I'll play a bit with it and reply to you.

**Comment by EugeneGirsh** [comment]
Hi!
I've learned about angular testing and jasmine. I've created a test site
with AngularJS and did my homework for testing with Jasmine.
Site: http://webzefat.azurewebsites.net
Test: http://webzefat.azurewebsites.net/test/specrunner.html
How can we proceed to a real work?
Cheers

**Comment by nmccready** [comment]
Get a good feel for 
- [bootstrap.coffee](https://github.com/angular-ui/angular-google-maps/blob/master/spec/coffee/bootstrap.coffee)
- [Helpers](https://github.com/angular-ui/angular-google-maps/blob/master/spec/coffee/helpers)
  - [initiator](https://github.com/angular-ui/angular-google-maps/blob/master/spec/coffee/helpers/initiator.coffee)
  - [google-mocks](https://github.com/angular-ui/angular-google-maps/blob/master/spec/coffee/helpers/google-api-mock.coffee)

- Grunt
  - [Jasmine Setup](https://github.com/angular-ui/angular-google-maps/blob/master/grunt/jasmine.coffee)

Run the coverage reports and see what is lacking. Also read comments in the specs for TODOs and just dig in.

@cthrax any other suggestions?

**Comment by cthrax** [comment]
No, the google-mocks I think is the most important piece to understand in this codebase, in regards to testing. As far as background knowledge, knowing how to manually compile a directive for testing purposes is useful, also make sure you know how to use $httpBackend.
